Content.Server/Mind/MindSystem.cs

```diff
@@ -90,7 +90,7 @@ private void OnShutdown(EntityUid uid, MindComponent mind, ComponentShutdown arg
 
                     // Async this so that we don't throw if the grid we're on is being deleted.
                     var gridId = spawnPosition.GetGridId(EntityManager);
-                    if (!spawnPosition.IsValid(EntityManager) || gridId == GridId.Invalid || !_mapManager.GridExists(gridId))
+                    if (!spawnPosition.IsValid(EntityManager) || gridId == GridId.Invalid || !_mapManager.GridExists(gridId) || spawnPosition.Position == Vector2.Zero)
```

Comment:
Wouldn't it just be easier to make the spawn attach them to the grid?

Reply:
You're right. I've updated it with this change.